internal/handlers/common/distinct.go

```diff
@@ -49,6 +49,13 @@ func GetDistinctParams(document *types.Document, l *zap.Logger) (*DistinctParams
 		return nil, err
 	}
 
+	switch dp.Filter.(type) {
```

Comment:
Could you add `default` case for `Filter` that is not a document or null please? Also could you add an integration test for non document/null filter please? If it returns error, we want it in integration test (not compat), if it doesn't return error we want it in compat test :hugs: